delta/plugins/elasticsearch/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/routes/ElasticSearchViewsDirectivesSpec.scala

```diff
@@ -68,14 +82,45 @@ class ElasticSearchViewsDirectivesSpec
       val bob      = User("bob", Label.unsafe("myrealm"))
       val bobId    = UrlUtils.encode(bob.asIri.toString)
 
-      Get(
-        s"/search/org/project?id=myId&deprecated=false&rev=2&createdBy=$aliciaId&updatedBy=$bobId&type=A&type=B&type=-C&schema=C&q=something"
-      ) ~> Accept(`*/*`) ~> route ~> check {
-        response.asString shouldEqual s"'${base}myId','false','2','$alicia','$bob','IncludedType(${vocab}A)|IncludedType(${vocab}B)|ExcludedType(${vocab}C)','${base}C','something'"
+      val query    = List(
+        "locate"     -> "self",
+        "id"         -> "myId",
+        "deprecated" -> "false",
+        "rev"        -> "2",
+        "createdBy"  -> aliciaId,
+        "updatedBy"  -> bobId,
+        "rev"        -> "2",
+        "type"       -> "A",
+        "type"       -> "B",
+        "type"       -> "-C",
+        "schema"     -> "mySchema",
+        "q"          -> "something"
+      ).map { case (k, v) => s"$k=$v" }.mkString("&")
+
+      val expected = ResourcesSearchParams(
+        locate = Some(iri"${base}self"),
+        id = Some(iri"${base}myId"),
+        deprecated = Some(false),
+        rev = Some(2),
+        createdBy = Some(alicia),
+        updatedBy = Some(bob),
+        types = List(
+          IncludedType(iri"${vocab}A"),
+          IncludedType(iri"${vocab}B"),
+          ExcludedType(iri"${vocab}C")
+        ),
+        schema = Some(ResourceRef.Latest(iri"${base}mySchema")),
+        q = Some("something")
+      )
+
+      Get(s"/search/org/project?$query") ~> Accept(`*/*`) ~> route ~> check {
+        response.as[ResourcesSearchParams] shouldEqual expected
```

Comment:
I think if you parse with an internal class then you won't catch breaking output changes

Reply:
You can achieve something similar by making a function which constructs the expected CSV

Reply:
The class is not internal, only the codec is.
The class contains options and an array so CSV does not really fit.
Besides, it would be better to keep the door leading to CSV closed and locked, we already have enough formats to support.

Reply:
`ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model.ResourcesSearchParams` is this not a class used by the server?

I'm confused because the previous test was asserting that the result was a CSV. Are you getting back a different response type now?

Reply:
Sorry, I thought you meant internal to the test class.
Here, the test is about test the directive, not the route.
The approach when testing a directive is to insert it in a bogus route which exists only for this test.
The CSV choice was only made in the context of the text.
I changed that for a json with circe codecs because it makes things clearer for a data structure with options and a list